### ISSUE TITLE
CoolProp won't install on Python

### ISSUE THREAD

**Issue by Knightroglycerin** [CoolProp/CoolProp#1432]
Hi everyone,

I'm a college student who is trying to install CoolProp for Python in order to complete a school project. However, when I put in `pip install CoolProp` I get an error message saying `SyntaxError: invalid syntax.` This happens in both the Python 3.6 IDLE and Python 2.7 running in a terminal window. My computer is a **MacBook Air** running **OS Sierra version 10.12.3**. Is there anything I need to download in order to install CoolProp? I have little programming experience outside of basic MatLab skills, so I might not have installed everything I need.

I have already installed CoolProp for MatLab, but the Python version can create graphs, which would be very helpful for my project. If there is a way to get the graphs in MatLab, that would be fine too.

 Any help will be appreciated.

Thanks,
Knightroglycerin

**Comment by ibell** [comment]
Hello, the pip install .... command should be run at the windows shell, not
within a shell in spyder.  To open a windows shell, go to the start command
in windows, then type "cmd", and press enter

**Comment by thorade** [comment]
On a Mac, hit the keys  ⌘ and space, then start to type `terminal` and hit enter.
In that terminal, enter `pip --version` and if it works, run `pip install -U coolprop`

**Comment by Knightroglycerin** [comment]
Hi everyone,

Thanks for your help, and sorry for relying so late. I've done what thorade recommended, but I keep getting error messages from Terminal. I tried it several times with similar results. Here's the error message:

Collecting coolprop
  Using cached CoolProp-6.1.0-cp27-cp27m-macosx_10_5_x86_64.whl
Installing collected packages: coolprop
Exception:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/wheel.py", line 345, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/wheel.py", line 316, in clobber
    ensure_dir(destdir)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/utils/__init__.py", line 83, in ensure_dir
    os.makedirs(path)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 13] Permission denied: '/Library/Python/2.7/site-packages/CoolProp'

I did try running CoolProp in Python just to be sure, and it didn't work. Python tells me there is no module named CoolProp. Is there any way to fix this error?

Thanks,
Knightroglycerin

**Comment by thorade** [comment]
Good to hear the first step worked. The error you get is : Permission denied to make/create a new directory. This can probably be solved by running the command with higher rights.
This is, I would say, not a problem of CoolProp, but of your Python distribution, and I am 99,9% sure someone else had the same problem before. A good resource to ask such very general questions is stackoverflow: http://stackoverflow.com/questions/tagged/python
But remember to google the error message before asking!

Another place to ask CoolProp related questions is the CoolProp mailing list, it is here:
https://groups.google.com/d/forum/coolprop-users
That link should probably be placed more prominently in the ReadMe.

Then, as you seem to be new to Python, you should be aware that there is Python2 and 3, I would highly recommend using Python3 for all new code, please also read this:
https://wiki.python.org/moin/Python2orPython3
Python 3 is more powerful and more clear at the same time.

And then, there is the Python from python.org, often called vanilla python (without additions), and in contrast there are distributions like Anaconda, that already include tons of packages.
https://www.continuum.io/downloads
For a quick start, such a distribution might be easier.

And, finally, you can mark code or error messages as code, read this guide:
https://guides.github.com/features/mastering-markdown/

**Comment by Knightroglycerin** [comment]
Thanks for your help! I'll look into it and see what I can find.

**Comment by ibell** [comment]
As always, I would recommend that you use the Anaconda python system:
https://www.continuum.io/downloads.  That is far preferable to using the
system python installation.

**Comment by Knightroglycerin** [comment]
Hi everyone,

It's been a while, but good news! I finally got around to looking up the problem, and I used 'pip install --user CoolProp' to install CoolProp in Terminal. Terminal says it successfully installed CoolProp, but when I try to run it in Python 3.6, it still won't run. Here is the code I tried to run; it's supposed to draw a simple plot:

import CoolProp
from CoolProp.Plots import PropertyPlot
plot = PropertyPlot('R290', 'ph')
plot.calc_isolines()
plot.show()

And this is the error message that Python returned: 

Traceback (most recent call last):
  File "/Users/JohnDoe/Documents/PythonPrograms/PropertyPlot.py", line 1, in <module>
    import CoolProp
ModuleNotFoundError: No module named 'CoolProp'

Is there anything I could do to fix this? I looked it up and there doesn't seem to be one definite solution. And also, since both of you have recommended Anaconda to me, would CoolProp run better in Anaconda than in Python 3.6? If so, I'll probably install Anaconda and use it instead.

Again, thanks for all your help. I'm not really a computer person, as I've said before, but everyone's replies have been very valuable in gradually getting me through this problem.

-Knightroglycerin

**Comment by ibell** [comment]
Can you use python 3.5?  I use that all the time without any problems.  It
installs python from wheels from pip too if you install from the
development server.

I think we have binaries building now, so try the second link from here:
http://www.coolprop.org/coolprop/wrappers/Python/index.html#automatic-installation

Please post your output from the install


On Apr 14, 2017 1:28 AM, "Knightroglycerin" <notifications@github.com>
wrote:

Hi everyone,

It's been a while, but good news! I finally got around to looking up the
problem, and I used 'pip install --user CoolProp' to install CoolProp in
Terminal. Terminal says it successfully installed CoolProp, but when I try
to run it in Python 3.6, it still won't run. Here is the code I tried to
run; it's supposed to draw a simple plot:

import CoolProp
from CoolProp.Plots import PropertyPlot
plot = PropertyPlot('R290', 'ph')
plot.calc_isolines()
plot.show()

And this is the error message that Python returned:

Traceback (most recent call last):
File "/Users/JohnDoe/Documents/PythonPrograms/PropertyPlot.py", line 1, in
import CoolProp
ModuleNotFoundError: No module named 'CoolProp'

Is there anything I could do to fix this? I looked it up and there doesn't
seem to be one definite solution. And also, since both of you have
recommended Anaconda to me, would CoolProp run better in Anaconda than in
Python 3.6? If so, I'll probably install Anaconda and use it instead.

Again, thanks for all your help. I'm not really a computer person, as I've
said before, but everyone's replies have been very valuable in gradually
getting me through this problem.

-Knightroglycerin

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/CoolProp/CoolProp/issues/1432#issuecomment-294107218>,
or mute the thread

**Comment by Knightroglycerin** [comment]
I tried the second link and it downloads just fine, but when I try to open it, my computer is bringing up an error message saying 'There is no application set to open the document “CoolProp-6.1.1.dev0-cp33-cp33m-win32.whl”'. I also tried the link right above it (the other one released on 2017-04-10) but my 7-zip unpacking program says it's corrupted.

For my previous install through Terminal, I can't remember the exact message but it was something like "Successfully installed CoolProp 6.1.0" or something along those lines. I could maybe try again to see the exact message but I don't know if that will leave two redundant copies of CoolProp somewhere in my computer.

Since it's recommended, I think I'll try installing Anaconda to see if that will run CoolProp. The Mac OS tab has download links for a Graphical Installer and a Command Line Installer; should I download both of them or just one?
